contracts/helpers/RebalancerFlashloan.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.8.19;
+
+import "./Rebalancer.sol";
+import { IERC4626 } from "interfaces/external/IERC4626.sol";
+import { IERC3156FlashBorrower } from "oz/interfaces/IERC3156FlashBorrower.sol";
+import { IERC3156FlashLender } from "oz/interfaces/IERC3156FlashLender.sol";
+
+/// @title RebalancerFlashloan
+/// @author Angle Labs, Inc.
+/// @dev Rebalancer contract for a Transmuter with as collaterals a liquid stablecoin and an ERC4626 token
+/// using this liquid stablecoin as an asset
+contract RebalancerFlashloan is Rebalancer, IERC3156FlashBorrower {
+    using SafeERC20 for IERC20;
+    using SafeCast for uint256;
+    bytes32 public constant CALLBACK_SUCCESS = keccak256("ERC3156FlashBorrower.onFlashLoan");
+
+    /// @notice ERC4626 Vault accepted as a collateral
+    IERC4626 public immutable VAULT;
+
+    /// @notice Liquid collateral wrapped in the vault
+    IERC20 public immutable COLLATERAL;
+
+    /// @notice Angle stablecoin flashloan contract
+    IERC3156FlashLender public immutable FLASHLOAN;
+
+    constructor(
+        IAccessControlManager _accessControlManager,
+        ITransmuter _transmuter,
+        IERC4626 _vault,
+        IERC3156FlashLender _flashloan
+    ) Rebalancer(_accessControlManager, _transmuter) {
+        if (address(_flashloan) == address(0)) revert ZeroAddress();
+        VAULT = _vault;
+        COLLATERAL = IERC20(_vault.asset());
+        FLASHLOAN = _flashloan;
+        COLLATERAL.safeApprove(address(_vault), type(uint256).max);
+        IERC20(AGTOKEN).safeApprove(address(_flashloan), type(uint256).max);
+    }
+
+    /// @notice Burns `amountStablecoins` for one collateral asset and mints stablecoins from the proceeds of the
+    /// first burn
+    /// @dev If `increase` is 1, then the system tries to increase its exposure to the yield bearing asset which means
+    /// burning stablecoin for the liquid asset, depositing into the ERC4626 vault, then minting the stablecoin
+    function adjustYieldExposure(uint256 amountStablecoins, uint8 increase) external {
+        if (!TRANSMUTER.isTrustedSeller(msg.sender)) revert NotTrusted();
```

Comment:
Why do we make this one only trutesSeller while other order from `Rabalancer` are onlyGuardian?

Either one or the other for both